### DISCUSSION TITLE
Need help sending document using standalone distribution

### DISCUSSION THREAD

**Discussion by alfchristensen** [OxalisCommunity/Oxalis-AS4#209] [Q&A] [answered]
Hi! 

### Context and what I have so far:
I have built a docker container from the norstella/oxalis:latest-11 image on dockerhub. The container runs as expected, and gives expected responses "Hello AS4 world" for the http://localhost:8080/as4. The container has its own directory with the certificate.pem  document.xml  oxalis-keystore.jks  oxalis-standalone.jar  oxalis.conf  truststore.jks. The /oxalis/conf has the following files: oxalis.conf, oxalis-keystore.jks, truststore.jks and logback.xml.

My oxalis.conf file looks like this:
oxalis.keystore {
  path=/oxalis/conf/oxalis-keystore.jks
  password="********"
  key.alias="cert"
  key.password="********"
}
oxalis.truststore {
  path=/oxalis/conf/truststore.jks
  password="********"
}

oxalis.logging.config=/oxalis/conf/logback.xml

transport.as4_peppol_v2 {
    profile = peppol-transport-as4-v2_0
    sender = oxalis-as4
    weight = 5000
}

### My problem:
When I try to run the standalone jar file with arguments to return the test document to testbed, I get the following error message in the terminal: "  No Content-Type header in response, probably a server error". The log shows this: "WARNING: Interceptor for has thrown exception, unwinding now
org.apache.cxf.binding.soap.SoapFault: Error reading XMLStreamReader: Unexpected character '-' (code 45) in prolog; expected '<'  "

For more details, please se the attached container log file and terminal log.

I have tried the following commands for running the jar file:
java -jar oxalis-standalone.jar -f document.xml -u http://localhost:8080/as4 -cert cert.pem -s 0192:810418052 -r 0088:TBCNTRL00002
java -jar oxalis-standalone.jar -f document.xml -u http://localhost:8080/as4 -cert cert.pem

Any help on this would be greatly appreciated!

Best Regards
Alf Christensen
[container_log.txt](https://github.com/OxalisCommunity/Oxalis-AS4/files/11618329/container_log.txt)
[terminal_log.txt](https://github.com/OxalisCommunity/Oxalis-AS4/files/11618330/terminal_log.txt)

**Comment by aaron-kumar** [comment]
@alfchristensen In general there are multiple problem. But let me first highlight main problem. 

There is problem with standalone command that you are using. 
1. You are missing Oxalis-AS4 library in classpath. This is reason why it is failing and you are getting error message : <code>org.apache.cxf.binding.soap.SoapFault: Error reading XMLStreamReader: Unexpected character '-' (code 45) in prolog; expected '<' </code> . You are trying to send AS2 message to AS4 endpoint. If you are missing Oxalis-AS4 library in classpath, then default protocol is "AS2"
2. When performing SML/SMP lookup (_which is the case in Production and Testbed_), following command will be used <code> java -cp "oxalis-distribution-6.0.0-RC3-distro/bin/*;oxalis-as4-6.0.0-RC3-dist/*" eu.sendregning.oxalis.Main -f document.xml </code>. **When returning document to Testbed**, you have to perform proper SML/SMP lookup, so use this option. Point-to-Point command used by you will not send document to Peppol Testbed. 
3. If you are sending point-to-point e.g. to yourselves for Test purpose then following could be command  <code> java -cp "oxalis-distribution-6.0.0-RC3-distro/bin/*;oxalis-as4-6.0.0-RC3-dist/*" eu.sendregning.oxalis.Main -f document.xml -u http://localhost:8080/as4 -cert cert.pem -s 0192:810418052 -r 0088:TBCNTRL00002 --protocol peppol-transport-as4-v2_0 </code>

Make sure that you have "oxalis distribution" version and "Oxalis-AS4" version available in path which you can download from https://github.com/OxalisCommunity/oxalis/releases & https://github.com/OxalisCommunity/Oxalis-AS4/releases 
SBD Header in "document.xml" contain routing information so you need to specify additional parameters explicitly when performing SML/SMP lookup. Put "0088:TBCNTRL00002"  as Receiver participantID in SBDH. 

**Note**: Modify above commands with the Oxalis & Oxalis-AS4 version that you are using.

**Reply by alfchristensen** [reply]
Hello again  @aaron-kumar, and thanks for your quick reply!

I have tried doing things exactly as you have explained. Created a folder the folder structure, and downloaded the files and folders from your reply. I am not able to execute the java command with classpath. I only get the following error message: 
"Error: Could not find or load main class eu.sendregning.oxalis.Main
Caused by: java.lang.ClassNotFoundException: eu.sendregning.oxalis.Main!

Please se my attached terminal log. All the necessary files should be present, I have triple checked the directory names, but I can't get around the error message. 
[terminal-log.txt](https://github.com/OxalisCommunity/Oxalis-AS4/files/11626787/terminal-log.txt)

Do you see anything I am doing wrong?

**Reply by aaron-kumar** [reply, marked as answer]
Since you are running command in Linux, so replace ";" with ":" . Important difference between setting Classpath in Windows and Linux is path separator which is ";" (semi-colon) in Windows and ":" (colon) in Linux

So for linux command will be:
java -cp "oxalis-distribution-6.0.0-RC3-distro/bin/*:oxalis-as4-6.0.0-RC3-dist/*" eu.sendregning.oxalis.Main -f document.xml

and I hope you have "oxalis-distribution-6.0.0-RC3-distro" and "oxalis-as4-6.0.0-RC3-dist" where you running this command. If not then use Full path.

I also notice that * is being removed by Github after "...bin/" and "......-dist/" . So place * after those to represent all libraries inside it.

**Reply by alfchristensen** [reply]
Thats it! Now it is sending! Thanks a lot @aaron-kumar !

**Comment by dladlk** [comment]
You are sending via AS2 protocol, not AS4. 

You can see it in your terminal.log:

Caused by: network.oxalis.api.lang.OxalisTransmissionException: No Content-Type header in response, probably a server error.
        at network.oxalis.**as2.outbound.As2MessageSender**.handleResponse(As2MessageSender.java:285)
        at network.oxalis.as2.outbound.As2MessageSender.sendHttpRequest(As2MessageSender.java:245)
        at network.oxalis.as2.outbound.As2MessageSender.send(As2MessageSender.java:154)
        at network.oxalis.as2.outbound.As2MessageSenderFacade.send(As2MessageSenderFacade.java:58)

Try to add a protocol into command line.
